### PR TITLE
[Master] - Added Custom login portal documentation 

### PR DESCRIPTION
**Pull request by divyaamunugama**
## Purpose
$subject

Related Issue: https://github.com/wso2/product-is/issues/14919

### CODE REVIEW

File: en/docs/references/extend/rebranding/customizable-login-portal.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
+# Customize login portal
+
+This feature can be used for both layout customization and branding customization of user interfaces like the login page, register page, username and password recovery pages, and single sign-on pages.
```
Comment (nilminiwso2) [inline en/docs/references/extend/rebranding/customizable-login-portal.md:3]:
```suggestion
You can customize the layout and branding of user interfaces like the login page, register page, username and password recovery pages, and single sign-on pages.
```

File: en/docs/references/extend/rebranding/customizable-login-portal.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+All UIs are available in the `authenticationendpoint`, `accountrecoveryendpoint`, and, `x509certificateauthenticationendpoint` apps located in `<IS_HOME>/repository/deployment/server/webapps` folder.
```
Comment (nilminiwso2) [inline en/docs/references/extend/rebranding/customizable-login-portal.md:5]:
```suggestion
All UIs are available in the `authenticationendpoint`, `accountrecoveryendpoint`, and `x509certificateauthenticationendpoint` apps located in `<IS_HOME>/repository/deployment/server/webapps` folder.
```

File: en/docs/references/extend/rebranding/customizable-login-portal.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+## General components of the UI
+
+All pages of the identity server are separated into three general components as shown below. You can reference these components when you write the custom layout using a special notation.
```
Comment (nilminiwso2) [inline en/docs/references/extend/rebranding/customizable-login-portal.md:9]:
```suggestion
All pages of WSO2 Identity Server are separated into three general components as shown below. You can reference these components when you write the custom layout using a special notation.
```

File: en/docs/references/extend/rebranding/customizable-login-portal.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+This section includes details about the special syntax that can be used when writing a custom layout code.
+
+- **Condition block**
+    
+    ![Condition Block]({{base_path}}/assets/img/references/condition-block.png)
+
+    This is similar to an `IF` block. This condition block will activate or deactivate according to the value of the `isPolicyPage` variable.
+    
+    - If the value of `isPolicyPage` is `true` or any non-empty string then the content inside the block will be executed.
```
Comment (nilminiwso2) [inline en/docs/references/extend/rebranding/customizable-login-portal.md:28]:
```suggestion
    - If the value of `isPolicyPage` is `true` or any non-empty string, then the content inside the block will be executed.
```

File: en/docs/references/extend/rebranding/customizable-login-portal.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+All pages of the identity server are separated into three general components as shown below. You can reference these components when you write the custom layout using a special notation.
+    
+- Login page
+    ![Login Page]({{base_path}}/assets/img/references/login-page-labelled.png)
+
+- Register page
+    ![Register Page]({{base_path}}/assets/img/references/register-page-labelled.png)
+
+
+## Layout Syntaxes
```
Comment (nilminiwso2) [inline en/docs/references/extend/rebranding/customizable-login-portal.md:18]:
```suggestion
## Layout syntaxes
```

File: en/docs/references/extend/rebranding/customizable-login-portal.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    This syntax will be used to indicate the position of the general component in the custom layout code. The component syntaxes will be replaced with actual content at runtime.
+
+- **Data Syntax**
+    
+    ![Data Syntax]({{base_path}}/assets/img/references/data-syntax.png)
+
+    This syntax can be used to add data to the layout code. The value stored in the `containerSize` variable will be converted to a string and placed in the corresponding location at runtime. All data syntax values will be sanitized before adding to the layout code.
+
+!!! info
+    All these variables will be provided by `JSP` pages. To get more data from JSP pages, add the data into the `layoutData` Map object from each JSP page. This `layoutData` map object can be found on each JSP page.
```
Comment (nilminiwso2) [inline en/docs/references/extend/rebranding/customizable-login-portal.md:53]:
```suggestion
    All these variables will be provided by `JSP` pages. To get more data from JSP pages, add the data into the `layoutData` map object from each JSP page. This `layoutData` map object can be found on each JSP page.
```

File: en/docs/references/extend/rebranding/customizable-login-portal.md [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+
+
+
+## Add a custom layout for the authentication/recovery pages
+
+To customize these pages, follow the steps below.
+
+!!! note
+    On this page, we will be referring to `<IS_HOME>/repository/deployment/server/webapps/` as `webapps` folder.
```
Comment (nilminiwso2) [inline en/docs/references/extend/rebranding/customizable-login-portal.md:62]:
```suggestion
    On this page, we will be referring to `<IS_HOME>/repository/deployment/server/webapps/` as the `webapps` folder.
```

File: en/docs/references/extend/rebranding/customizable-login-portal.md [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+Navigate to `webapps/authenticationendpoint` and create a new folder named `extensions`.
+
+!!! note
+    Repeat the above step for `accountrecoveryendpoint` and `x509certificateauthenticationendpoint` as well.
+
+### Step 2: Create the basic custom layout resources
+
+To create the basic custom layout resources:
+
+1. On the `authenticationendpoint` folder, navigate to the created `extensions` folder and create `layouts` folder and `customs` folders recursively.
```
Comment (nilminiwso2) [inline en/docs/references/extend/rebranding/customizable-login-portal.md:75]:
```suggestion
1. On the `authenticationendpoint` folder, navigate to the created `extensions` folder and create the `layouts` folder and the `customs` folders recursively.
```

File: en/docs/references/extend/rebranding/customizable-login-portal.md [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+    
+    ```
+    └── extensions
+        └── layouts
+            └── custom
+    ```
+
+3. Navigate to `webapps/authenticationendpoint/includes/layouts/default` folder, copy the `body.html` file and add it to the created `custom` folder.
```
Comment (nilminiwso2) [inline en/docs/references/extend/rebranding/customizable-login-portal.md:83]:
```suggestion
3. Navigate to the `webapps/authenticationendpoint/includes/layouts/default` folder, copy the `body.html` file, and add it to the created `custom` folder.
```

File: en/docs/references/extend/rebranding/customizable-login-portal.md [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+    - `styles.css` file
+    - `script.js` file
+    - `assets` folder
+
+!!! note
+    Repeat the above steps for `accountrecoveryendpoint` and `x509certificateauthenticationendpoint` as well.
+
+### Step 3: Add a custom layout design
+
+To add a custom layout to the login page
```
Comment (nilminiwso2) [inline en/docs/references/extend/rebranding/customizable-login-portal.md:95]:
```suggestion
To add a custom layout to the login page:
```

File: en/docs/references/extend/rebranding/customizable-login-portal.md [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+
+1. Navigate to `webapps/authenticationendpoint/extensions/layouts/custom/`
```
Comment (nilminiwso2) [inline en/docs/references/extend/rebranding/customizable-login-portal.md:97]: incomplete

File: en/docs/references/extend/rebranding/customizable-login-portal.md [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+
+    1. Copy the code snippet from the [html file](https://github.com/wso2/docs-is/tree/master/en/docs/assets/code-samples/body.html) into the `body.html` file.
+
+    2. Copy the code snippet from the [css file](https://github.com/wso2/docs-is/tree/master/en/docs/assets/code-samples/style.css) into the `styles.css` file.
+
+    3. Copy the code snippet from the [javascript file](https://github.com/wso2/docs-is/tree/master/en/docs/assets/code-samples/script.js) into the `script.js` file.
+
+    4. Copy the [illustraction.svg](https://github.com/wso2/docs-is/tree/master/en/docs/assets/code-samples/illustration.svg) file into the `assets` folder.
+
+2. Navigate to `webapps/authenticationendpoint/includes` directory and copy the `product-title.jsp` and `product-footer.jsp` files into the `webapps/authenticationendpoint/extensions` directory.
```
Comment (nilminiwso2) [inline en/docs/references/extend/rebranding/customizable-login-portal.md:107]:
```suggestion
2. Navigate to the `webapps/authenticationendpoint/includes` directory and copy the `product-title.jsp` and `product-footer.jsp` files to the `webapps/authenticationendpoint/extensions` directory.
```

File: en/docs/references/extend/rebranding/customizable-login-portal.md [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+
+    1. Copy content from the [`jsp` file containing the project title](https://github.com/wso2/docs-is/tree/master/en/docs/assets/code-samples/project-title-1.jsp) into the `webapps/authenticationendpoint/extensions/product-title.jsp` file of the app.
```
Comment (nilminiwso2) [inline en/docs/references/extend/rebranding/customizable-login-portal.md:109]:
```suggestion
    1. Copy content from the [`jsp` file containing the project title](https://github.com/wso2/docs-is/tree/master/en/docs/assets/code-samples/project-title-1.jsp) to the `webapps/authenticationendpoint/extensions/product-title.jsp` file of the app.
```

File: en/docs/references/extend/rebranding/customizable-login-portal.md [hunk 14]
```diff
@@ -0,0 +1,134 @@
...
+
+    2. Copy content from the [`jsp` file containing the project footer](https://github.com/wso2/docs-is/tree/master/en/docs/assets/code-samples/product-footer-1.jsp) into the `webapps/authenticationendpoint/extensions/product-footer.jsp` file of the app.
```
Comment (nilminiwso2) [inline en/docs/references/extend/rebranding/customizable-login-portal.md:111]:
```suggestion
    2. Copy content from the [`jsp` file containing the project footer](https://github.com/wso2/docs-is/tree/master/en/docs/assets/code-samples/product-footer-1.jsp) to the `webapps/authenticationendpoint/extensions/product-footer.jsp` file of the app.
```

File: en/docs/references/extend/rebranding/customizable-login-portal.md [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+3. Refresh the browser and check out the added custom layout.
+
+    !!! tip
+        In certain cases, the server might stop working after adding changes. In such a case, restart the server to check the changes.
+
+!!! note 
+    - Use the following **best practices** when creating a custom layout:
+        - Add a prefix for the new CSS classes so that the newly added classes will not conflict with existing classes.
+        - Set `cache="false"` when developing a custom layout. (Make sure to remove the `cache="false"` in production code).
+    - If you need to build the apps from `identity-apps` source code:
```
Comment (nilminiwso2) [inline en/docs/references/extend/rebranding/customizable-login-portal.md:132]:
```suggestion
    - If you need to build the apps from the `identity-apps` source code:
```

File: en/docs/references/extend/rebranding/customizable-login-portal.md [hunk 16]
```diff
@@ -0,0 +1,134 @@
...
+        1. Add the developed custom layout folder into `<IDENTITY_APPS_SOURCE_CODE>/components/login-portal-layouts/layouts` directory.
```
Comment (nilminiwso2) [inline en/docs/references/extend/rebranding/customizable-login-portal.md:133]:
```suggestion
        1. Add the developed custom layout folder into the `<IDENTITY_APPS_SOURCE_CODE>/components/login-portal-layouts/layouts` directory.
```

File: en/docs/references/extend/rebranding/customizable-login-portal.md [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+        - [Project title](https://github.com/wso2/docs-is/tree/master/en/docs/assets/code-samples/product-title-2.jsp).
+
+        - [Project footer](https://github.com/wso2/docs-is/tree/master/en/docs/assets/code-samples/product-footer-2.jsp)
+
+3. Refresh the browser and check out the added custom layout.
+
+    !!! tip
+        In certain cases, the server might stop working after adding changes. In such a case, restart the server to check the changes.
+
+!!! note
```
Comment (nilminiwso2) [inline en/docs/references/extend/rebranding/customizable-login-portal.md:128]: Two notes following one after the other.
Why not create another heading `Best practices` for this list.

File: en/docs/references/extend/rebranding/customizable-login-portal.md [hunk 18]
```diff
@@ -0,0 +1,134 @@
+# Customize login portal
```
Comment (nilminiwso2) [inline en/docs/references/extend/rebranding/customizable-login-portal.md:1]: Let's watch the consistency with `directory` and `folder`.

File: en/docs/references/extend/rebranding/customizable-login-portal.md (hunk 11 above)
Comment (divyaamunugama) [reply en/docs/references/extend/rebranding/customizable-login-portal.md:97]: `Navigate to `webapps/authenticationendpoint/extensions/layouts/custom/` and copy the content as instructed below:`

Is this okay? Sub-points 1,2,3,4 should be done while the user is in this folder

File: en/docs/references/extend/rebranding/customizable-login-portal.md (hunk 16 above)
Comment (divyaamunugama) [inline en/docs/references/extend/rebranding/customizable-login-portal.md]:
```suggestion
        1. Add the developed custom layout folder into `<IDENTITY_APPS_SOURCE_CODE>/components/login-portal-layouts/layouts` folder.
```

File: en/docs/references/extend/rebranding/customizable-login-portal.md [hunk 19]
```diff
@@ -0,0 +1,143 @@
+# Customize Login Portal
```
Comment (nilminiwso2) [inline en/docs/references/extend/rebranding/customizable-login-portal.md:1]:
```suggestion
# Customize Login Pages
```

File: en/mkdocs.yml [hunk 20]
```diff
@@ -664,7 +664,7 @@ nav:
         - Re-brand WSO2 Identity Server UIs: references/extend/rebranding/rebrand-identity-server-uis.md
         - Re-brand the Default Login Page: references/extend/rebranding/rebrand-login-pages-for-service-providers.md
         - Re-brand the SSO Redirection Page: references/extend/rebranding/rebrand-the-sso-redirection-page.md
-        #- Customize Login page: references/extend/rebranding/customizable-login-portal.md
+        - Customize Login page: references/extend/rebranding/customizable-login-portal.md
```
Comment (nilminiwso2) [inline en/mkdocs.yml:667]:
```suggestion
        - Customize Login Pages: references/extend/rebranding/customizable-login-portal.md
```

**Review by nilminiwso2 — APPROVED** [review]

**Pull request merged**